decidim-core/lib/tasks/decidim_webpacker_tasks.rake

```diff
@@ -79,46 +79,27 @@ namespace :decidim do
     end
 
     def decidim_npm_packages
-      case decidim_gemspec.source
-      when Bundler::Source::Path
-        gem_path = decidim_gemspec.source.path
-        gem_path = Pathname.new(ENV["BUNDLE_GEMFILE"]).dirname.join(gem_path) if gem_path.relative?
-
-        # The packages folder needs to be copied to the application folder
-        # because the linked dependencies are not installed when packages
-        # are installed using file references outside the application root
-        # where the `package.json` is located at. For more information, see:
-        # https://github.com/npm/cli/issues/2339
-        FileUtils.cp_r("#{gem_path}/packages", rails_app_path)
-
-        {
-          dev: "./packages/dev",
-          prod: "./packages/all"
-        }
-      when Bundler::Source::Rubygems
-        if decidim_gemspec.version.to_s =~ /\.dev$/
-          # With the .dev version the package does not exist at NPM yet.
-          {
-            dev: "https://gitpkg.now.sh/decidim/decidim/packages_dev/dev",
-            prod: "https://gitpkg.now.sh/decidim/decidim/packages_dev/all"
-          }
-        else
-          {
-            dev: "@decidim/dev@~#{decidim_gemspec.version}",
-            prod: "@decidim/all@~#{decidim_gemspec.version}"
-          }
-        end
-      when Bundler::Source::Git
-        github_repo =
-          decidim_gemspec.source.uri[%r{github\.com/([^/]*/[^.]*)}, 1] ||
-          "decidim/decidim"
-        branch = decidim_gemspec.source.branch
-
-        {
-          dev: "https://gitpkg.now.sh/#{github_repo}/packages_dev/dev?#{branch}",
-          prod: "https://gitpkg.now.sh/#{github_repo}/packages_dev/all?#{branch}"
+      if decidim_gemspec.source.is_a?(Bundler::Source::Rubygems)
+        return {
+          dev: "@decidim/dev@~#{decidim_gemspec.version}",
+          prod: "@decidim/all@~#{decidim_gemspec.version}"
         }
       end
+
+      gem_path = decidim_gemspec.source.path
+      gem_path = Pathname.new(ENV["BUNDLE_GEMFILE"]).dirname.join(gem_path) if gem_path.relative?
+
+      # The packages folder needs to be copied to the application folder
+      # because the linked dependencies are not installed when packages
+      # are installed using file references outside the application root
+      # where the `package.json` is located at. For more information, see:
+      # https://github.com/npm/cli/issues/2339
+      FileUtils.cp_r("#{gem_path}/packages", rails_app_path)
```

Comment:
If the gems doesn't come from rubygems, just copy the `packages` folder from the gem's folder into the app and install it locally. What do you think about this approach?

Reply:
@leio10 I think we discussed this approach already at https://github.com/decidim/decidim/pull/8119#issuecomment-859465289.

I think copying the `packages` folder is justified only when the gem source is a path. If the gems come from GitHub, we should install the NPM packages also from GitHub.

Reply:
I agree with you, but I don't find a better way to do it when having references to other packages:

* Using local references doesn't work.
* Using `gitpkg` is not trustworthy. If you install the gem using the commit `123` you need to be sure that all the other packages are from the same commit. That's why I think that copying the folder is the less worst solution.

Reply:
In my head, I would separate the NPM packages to live their own life compared to the Decidim gems.

So you would never want to reference a single commit from the Decidim repository or Decidim fork for the NPM packages. I think if you want to customize the NPM packages, you need to make a fork of them yourself and reference that fork where you have updated all the references inside the different packages.

I would think that modifying the NPM package's in so much detail that you would have to reference a single commit shouldn't be necessary most times, at least for the time being. They don't contain any running front-end code for Decidim right now, they are mostly for the "essentials" required to manage the packages with Webpacker.